bindings/haskell/haskell-src/OpenDAL.hs

```diff
@@ -28,16 +33,48 @@ import Data.HashMap.Strict (HashMap)
 import qualified Data.HashMap.Strict as HashMap
 import Foreign
 import Foreign.C.String
+import Foreign.C.Types (CChar)
 import OpenDAL.FFI
 
-newtype Operator = Operator (Ptr RawOperator)
+newtype Operator = Operator (ForeignPtr RawOperator)
+
+data ErrorCode
+  = FFIError
+  | Unexpected
+  | Unsupported
+  | ConfigInvalid
+  | NotFound
+  | PermissionDenied
+  | IsADirectory
+  | NotADirectory
+  | AlreadyExists
+  | RateLimited
+  | IsSameFile
+  deriving (Eq, Show)
+
+data OpenDALError = OpenDALError {errorCode :: ErrorCode, message :: String}
```

Comment:
I prefer to not add `OpenDAL` prefix for struct in binding. Is `Error` reserved in hashell?

Reply:
We can rename it to `Error`. However, the standard library has a general type class (similar to `trait` in Rust) called `Error`. This is prone to naming clashes.

I prefer to use `OpenDALError` here.